arbitrum-docs/run-arbitrum-node/run-nitro-dev-node.mdx

```diff
@@ -0,0 +1,168 @@
+
+## What the `--dev` flag does
+
+The `--dev` flag simplifies the setup by initializing a development environment with a number of preset parameters, creating an L2-only chain without requiring an L1 listener. When you use the `--dev` flag, Nitro automatically applies the following parameters:
+
+In `--dev` mode, a default development account is available, which is pre-funded with ETH in all networks. This account can be used for deploying contracts, interacting with the chain, and assuming chain ownership.
+
+- **Address:** 0x3f1Eae7D46d88F08fc2F8ed27FCb2AB183EB2d0E
+- **Private key:** 0xb6b15c8cb491557369f3c7d2c287b053eb229daa9c22138887752191c9520659
+
+## Setting the chain owner with `ArbDebug` precompile
+
+In Nitro `--dev` mode, the default chain owner is set to `0x0000000000000000000000000000000000000000`. However, you can use the [`ArbDebug` precompile](https://github.com/OffchainLabs/nitro-contracts/blob/main/src/precompiles/ArbDebug.sol) to set the chain owner. This precompile includes [the `becomeChainOwner()` function](https://github.com/OffchainLabs/nitro-contracts/blob/fbbcef09c95f69decabaced3da683f987902f3e2/src/precompiles/ArbDebug.sol#L13), which can be called to assume ownership of the chain. Chain ownership is important because it allows the owner to perform certain critical functions within the Arbitrum environment.
+
+To set the chain owner, you can call the `becomeChainOwner()` function with your preffered private key, we'll use the dev account as an example:
+
+```bash
+cast send 0x00000000000000000000000000000000000000FF "becomeChainOwner()" --private-key 0xb6b15c8cb491557369f3c7d2c287b053eb229daa9c22138887752191c9520659 --rpc-url http://127.0.0.1:8547
+```
+
+- `0x00000000000000000000000000000000000000FF`: This is the address of the ArbDebug precompile on Arbitrum.
+- `becomeChainOwner()`: This is the function call that transfers ownership of the chain to the caller.
+- `--private-key`: Use the private key of the account that should become the chain owner.
+- `--rpc-url http://127.0.0.1:8547`: Ensure that this points to your local Nitro dev node running in `--dev` mode.
+
+Once you're the chain owner, you'll be able to access various Arbitrum-specific functions, such as:
+
+- Managing the Sequencer
+- Modifying chain parameters
+- Accessing debugging features
+
+This functionality is especially useful for development and testing purposes when using Nitro in `--dev` mode.
+
+## Running Nitro dev node for Stylus testing
+
+If your goal is to test Stylus contracts on Nitro in `--dev` mode, you can set up the environment in the same way, but with the added step of deploying the `Stylus Cache Manager`, which is necessary for Stylus-specific functionality.
+
+To automate this process, you can use a bash script that both runs the Nitro dev node and deploys the `Stylus Cache Manager` contract automatically. Here's how you can do it:
+
+### Step 1: Create a Bash script to automate the process
```

Comment:
I'm thinking it might be better to just have this script stored in a repo and recommend it for the primary way to run the testnode. If we did that we could just store it in the testnode repo potentially (or make a separate devnode) repo and then have the primary instructions be cloning and running that parallel to the testnode instructions

Reply:
That sounds neat! I'm more leaning towards to creating a separate devnode repo for the sake of clarity.